collectors/proc.plugin/sys_class_drm.c

```diff
@@ -615,34 +615,46 @@ static void card_free(struct card *c){
 }
 
 static int check_card_is_amdgpu(const char *const pathname){
-    procfile *const ff = procfile_open(pathname, " ", PROCFILE_FLAG_NO_ERROR_ON_FILE_IO);
-    if(unlikely(!procfile_readall(ff))){
-        procfile_close(ff);
-        return 1; // error
+    int rc = -1;
```

Comment:
Should we not set `0` by default? If we do not need to se, so we can remove code from line `622`.
If you accept the suggestion, this means that we do not need to set value to zero at line `634`.

Reply:
I generally prefer to set `rc` to a failed code and then to `0` (where `0` means success), so that in case of a bug it fails gracefully.